### DISCUSSION TITLE
Module 6 Support

### DISCUSSION THREAD

**Discussion by travisz09** [ua-gist604b-f24/syllabus#35] [General]
Good morning all,

I know that everyone has been having a difficult time with module 6 over the last week. I have been struggling as well. For what it's worth, it is possible, I was able to complete assignment 6-2 earlier this morning. I will try to answer as many specific questions as I can, but I'm just as confused as many of you are. I think that some of it simply comes down to timing and good luck.

Http errors seem to be pretty common. If you are getting a 403 error, or "the white screen of death" while trying to access geoserver, my best advice is to walk away and try again later. Geoserver also seems to be pretty buggy in general. If something in geoserver is not working, you might have to try to find an alternate method.

I will try to help as many people as I can. Please send screenshots and be a specific as possible.

Best of luck,
Travis

P.S. I am using Firefox.

**Comment by jenw-ua** [comment]
Hi Travis, if you have a chance, can you help with 6.2? I'm just trying to save the updated osm store template in 6.2 step 7. It will not save. Aaryn mentions renaming the sld and running this work around, but I'm literally just confused now...
https://github.com/ua-gist604b-f24/syllabus/discussions/39
[https://github.com/ua-gist604b-f24/syllabus/discussions/39](url)

**Reply by travisz09** [reply]
I'm sorry @jenw-ua, I just saw your message a few moments ago.

If you are online now I can try to help you.

**Reply by travisz09** [reply]
Text me, and we can schedule a time to work on this. (360)915-3163.

**Reply by travisz09** [reply]
I'm in [https://us04web.zoom.us/j/6624203189?pwd=vdkBEg9wAqHbtQmUkmQSnYQa25b2xl.1](https://us04web.zoom.us/j/6624203189?pwd=vdkBEg9wAqHbtQmUkmQSnYQa25b2xl.1)

**Reply by jenw-ua** [reply]
Thanks for trying to help. Really appreciate it!

**Reply by travisz09** [reply]
OK. I think I may have found something here.

![image](https://github.com/user-attachments/assets/f5cd08a6-c6c1-436f-9aa3-efa29022b5bd)

**Reply by travisz09** [reply]
In the tutorial from the repository that we forked in step 1, near the bottom. It gives an option for "Parameterizing" from the command line.

**Reply by travisz09** [reply]
I don't understand the comment about the "Java vitural machine", but the line of code at the end looks like what you need.

**Reply by jenw-ua** [reply]
Should I try running that in the command line? Or restart codespace, run that line first? Or substitute that in? Just trying to figure out how to apply.

**Comment by travisz09** [comment]
I'm going to try this on my end, on a new codespace and see if it works for me. It might need a little tweeking to get right.

POSTGRES_ENDPOINT=127.0.0.1;POSTGRES_PORT=5432;POSTGRES_PASSWORD=postgres;POSTGRES_USER=postgres;OSM_DB=hawaii;OSM_SCHEMA=import

**Comment by travisz09** [comment]
OK! That ☝️ did not work, but I found another method that I think will work. It is somewhat involved, but I will walk through it.

**Reply by travisz09** [reply]
First, start a whole new codespace, clean slate.

**Reply by travisz09** [reply]
Clone the tutorial github repository according to step 1.

**Reply by travisz09** [reply]
Do not proceed to step 2.

**Reply by jenw-ua** [reply]
Okay! hang one min I'll try....

**Reply by travisz09** [reply]
In the file explorer, there is a directory osm-styles/workspaces/osm/osm/datastore.xml

**Reply by travisz09** [reply]
Open datastore.xml.

**Reply by jenw-ua** [reply]
Working...it's a little slow right now for some reason but still setting up codespace

**Reply by travisz09** [reply]
Search through the xml script for all the parameters mentioned in step 7. Carefully update each of the fields with the parameters for your db. Leave the "plain:" in the password field.

![image](https://github.com/user-attachments/assets/eb9ca51c-175d-4624-8d18-79957f8d2a93)

**Reply by travisz09** [reply]
Save the updated document (ctrl+s), just to be sure.

**Reply by travisz09** [reply]
Proceed normally for steps 2-6.

**Reply by jenw-ua** [reply]
Leave the $ and replace inside the { }  ? Just checking...

**Reply by travisz09** [reply]
Open the geoserver on port 8080, navigate to the osm store, **but do not make any changes!** Confirm that the parameters are set correctly, as shown in step 7.

**Reply by travisz09** [reply]
Replace everything, including the $ and { }.

**Reply by travisz09** [reply]
Example.
![image](https://github.com/user-attachments/assets/dd5864b5-8703-4e0f-b844-53423198a53b)

**Reply by jenw-ua** [reply]
<img width="697" alt="Screenshot 2024-11-27 at 5 30 04 PM" src="https://github.com/user-attachments/assets/044e8525-6415-4d2b-9b17-3a642295518d">

**Reply by travisz09** [reply]
After double-checking the osm store, navigate to the layer group osm:osm. Preview the layer group, zoom to hawaii. Take a screenshot and upload to your codespace. Complete the assignment as usual.

**Reply by jenw-ua** [reply]
Okay shoot, one min. Thanks.

**Reply by jenw-ua** [reply]
<img width="711" alt="Screenshot 2024-11-27 at 5 32 24 PM" src="https://github.com/user-attachments/assets/ae8799e7-3746-4c94-b4c2-175af31d8377">

**Reply by jenw-ua** [reply]
Just want to make sure it's not going to be a typo or something contributing. That password plain:postgres looks okay, etc?

**Reply by travisz09** [reply]
That appears to be correct. 🤞

**Reply by jenw-ua** [reply]
okay just finished step 6...

**Reply by jenw-ua** [reply]
So the miraculous thing is that the update to the stored has been "saved" now!  But I'm trying to see if the rest looks right...
<img width="853" alt="Screenshot 2024-11-27 at 5 44 21 PM" src="https://github.com/user-attachments/assets/df7bda0b-225e-4431-bda8-4e567c953211">

**Reply by travisz09** [reply]
That looks correct. Can you access the layer preview?

**Reply by jenw-ua** [reply]
Okay at quick glance it doesn't look like Hawaii, it looks like Iceland still?!

**Reply by jenw-ua** [reply]
But I've never made it this far into the assignment yet, so I'm not sure I'm doing the right thing. Here's my layergroup and I'm seeing icesheets...
<img width="1071" alt="Screenshot 2024-11-27 at 5 47 43 PM" src="https://github.com/user-attachments/assets/b325bcaf-7899-4a59-b68b-66c4c34895b8">

**Reply by travisz09** [reply]
Ignore that, that is just some low-res OSM layers. Sorry, I think I said to go to Layer Groups, you actually want Layer Preview.

**Reply by travisz09** [reply]
Click the OpenLayers link beside osm:osm to preview.
![image](https://github.com/user-attachments/assets/45f7e8c1-877a-4f63-979d-a44200128a1c)

**Reply by jenw-ua** [reply]
OMG I think it worked! One minute....zooming in like the video....

**Reply by jenw-ua** [reply]
YES, it worked!! Just a screenshot zoomed in to the data like the video now? I'm like, really, did this actually work? In shock. THANK YOU SO MUCH!!!

A+

**Reply by jenw-ua** [reply]
It actually makes sense what we did. Thank you so much for finding that information and walking me through it. Great job, seriously.

**Reply by travisz09** [reply]
You're welcome!

**Reply by jenw-ua** [reply]
Thanks again. Really appreciate your time! :)
